LSVMBUS.sh: fix scsi controller spread check on cores

chvalean opened the pull request:
Fixes #324

On a VM with 6 cores, scsi controller is spread on 2 cores. Handle this situation by checking the parity 4:1 better.

Other changes:
- Typo fixes and code indentation.
- Fix python detection compatibility with redhat 8.
- Display final results in test summary, not only LogMsg.

---

Test results:
SKU: Standard_NC6 - 6 cores

```
ARM Image Under Test  : RedHat : RHEL : 8 : 8.0.2019050711
ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
1 CORE                 LSVMBUS                                                                           PASS                 1.86
```

```
ARM Image Under Test  : Canonical : UbuntuServer : 19.04-DAILY : latest
ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes)
1 CORE                 LSVMBUS                                                                           PASS                 1.87
```

rmsivakanth commented:
```
Execution is verified.

11/01/2018 07:26:08 : [INFO ] ~~~~~~~~~~~~~~~TEST END : PERF-SYSCALL-BENCHMARK~~~~~~~~~~
11/01/2018 07:26:08 : [INFO ] CURRENT - PASS    - 1
11/01/2018 07:26:08 : [INFO ] CURRENT - FAIL    - 0
11/01/2018 07:26:08 : [INFO ] CURRENT - ABORTED - 0
11/01/2018 07:26:08 : [INFO ] Checking background cleanup jobs..
11/01/2018 07:26:08 : [INFO ] Removed 0 context files.
11/01/2018 07:26:08 : [INFO ] Cycle Finished.. TC-63662
11/01/2018 07:26:09 : [INFO ]
Test Results Summary
ICA test run on 11/01/2018 07:14:39
ARM Image under test OpenLogic : CentOS : 7.4 : latest
Total Executed TestCases        1 (       1 Pass,        0 Fail,        0 Abort)
Total Execution Time(dd:hh:mm) 0:0:11
XML file: TestConfiguration

#Sr. Test Name : Test Result : Test Duration (in minutes)
----------------------------------------------------
1. PERF-SYSCALL-BENCHMARK : PASS : 11.43


Logs can be found at .\TestResults\2018-01-11-07-14-31-5612


11/01/2018 07:26:09 : [INFO ] Creating 'Azure-63662-buildlogs.zip' from '.\TestResults\2018-01-11-07-14-31-5612'
11/01/2018 07:26:10 : [INFO ] R:\GitHub\LISAv2\Azure-63662-buildlogs.zip created successfully.
11/01/2018 07:26:10 : [INFO ] Copied : .\report\report_TC-63662.xml --> .\report\report_TC-63662-junit.xml
11/01/2018 07:26:10 : [INFO ] Analysing results..
11/01/2018 07:26:10 : [INFO ] PASS  : 1
11/01/2018 07:26:10 : [INFO ] FAIL  : 0
11/01/2018 07:26:10 : [INFO ] ABORT : 0
11/01/2018 07:26:10 : [INFO ] LISAv2 exits with code: 0
```

lizzha commented on Testscripts/Windows/PERF-SYSCALL-BENCHMARK.ps1:
```diff
@@ -127,8 +127,53 @@ collect_VM_properties
+                    }
+                    $SQLQuery = $SQLQuery.TrimEnd(',')
+                    LogMsg $SQLQuery
+                    $connection = New-Object System.Data.SqlClient.SqlConnection
```
use function UploadTestResultToDatabase in Framework.psm1

lizzha commented on Testscripts/Windows/PERF-SYSCALL-BENCHMARK.ps1:
```diff
@@ -127,8 +127,45 @@ collect_VM_properties
+                    if ($UseAzureResourceManager) {
+                        $HostType   = "Azure-ARM"
+                    } else {
+                        $HostType   = "Azure"
```
missed this one, as LISAv2 support ARM only, there's no need to set different value for HostType, set it to "Azure" only

juhlee-microsoft replied:
This comment is still valid, @rmsivakanth

rmsivakanth replied:
fixed.

lizzha commented:
Not to merge until the last comment is addressed

chvalean committed:
LSVMBUS.sh: fix scsi controller spread check on cores

Fixes #324

chvalean committed:
LSVMBUS.sh: fix ignored devices cleanup

Cleanup of ignored devices from lsvmbus.log was broken, as the sed was actually removing all lines after the matching string.
What was observed is that there might still be devices actually after that ignored one, so that was causing the referred issue where no NIC adapters where showing up in automation - lsvmbus tools output was correct.

Fixed the sed command by cleaning up from the matching ignored controller/adapter pattern, up to the next empty line, which is the split between devices.

chvalean committed:
LSVMBUS.sh: Fix compatibility with RedHat 8

chvalean committed:
Update LSVMBUS.sh
